### PR TITLE
Improves mining voucher choices and minebots

### PR DESCRIPTION
**Pull request by ChangelingRain**
:cl: Joan
tweak: The crusher kit now includes an advanced mining scanner.
tweak: The resonator kit now includes webbing and a small extinguisher.
tweak: The minebot kit now includes a minebot passthrough kinetic accelerator module, which will cause kinetic accelerator shots to pass through minebots. The welding goggles have been replaced with a welding helmet, allowing you to wear mesons and still be able to repair the minebot without eye damage.
feature: You can now install kinetic accelerator modkits on minebots. Some exceptions may apply. Crowbar to remove modkits.
balance: Minebots now shoot 33% faster by default(3 seconds to 2). The minebot cooldown upgrade still produces a fire rate of 1 second.
balance: Minebots are now slightly less likely to sit in melee like idiots, and are now healed for 15 instead of 10 when welded.
balance: Sentient minebots are penalized; they cannot have armor and melee upgrades installed, and making them sentient will override those upgrades if they were installed. In addition, they move very slightly slower and have their kinetic accelerator's cooldown increased by 1 second.
/:cl:

[the basic scanner sucks ass and nobody wants to use it, it's actually hilarious](https://atlantaned.space/statbus/stat.php?stat=mining_voucher_redeemed&year=2018&month=1)
also in those stats: minebot picked a mere 6 more times than the conscription kit, because it's an idiot drone, easy to accidentally hit, and did I mention how much it likes going into melee?

tbh it's quite possible the extra buffs aren't needed, just the passthrough module, but it's better to go a little too far rather than not far enough.

### CODE REVIEW

**Comment by Repukan** [comment]
Those buffs are too much.

Unless you make them mutually exclusive with sentience.

**Comment by Onule** [comment]
Yo, since this is a minebot change, do you want to use these sprites?
minebot3

Movement buff is too much, give them access to mining outpost EVA instead.

**Comment by 81Denton** [comment]
Idiot drone is an overstatement, that thing is a trashcan on wheels without sentience. Not to mention that in all of January, it's only been bought 146 times. Even luxury shelter capsules are more popular than the drone.

**Comment by PKPenguin321** [comment]
sentienced minebots are fucking stupid and this will make them even crazier which is a shame because AI minebots could use these buffs

**Comment by Xhuis** [comment]
At this point, I have to wonder if it'd be too power-creepy to buff the normal mining scanner to advanced levels and have the advanced one's images stick around longer or something.

**Comment by Bawhoppen** [comment]
Seems like creep. Shouldn't the basic beginner stuff be weak and less useful?

**Comment by Xhuis** [comment]
We're at a point where so few people use the basic stuff to begin with that most beginners take the advanced scanner anyway, if the numbers are to be trusted.

**Comment by JohnMcLaime** [comment]
Yeah what pkp said, also does kinetic pass module affect other projectiles?

**Comment by JohnMcLaime** [comment]
Welp, got confused.

> feature: You can now install kinetic accelerator modkits on minebots. Some exceptions may apply. Crowbar to remove modkits.

**Comment by improvedname** [comment]
Just make the ai upgrade module mining loot lol

File: code/game/objects/items/cards_ids.dm [hunk 1]
```diff
@@ -338,7 +338,7 @@ update_label("John Doe", "Clowny")
 
 /obj/item/card/id/mining
 	name = "mining ID"
-	access = list(ACCESS_MINING, ACCESS_MINING_STATION, ACCESS_MINERAL_STOREROOM)
+	access = list(ACCESS_MINING, ACCESS_MINING_STATION, ACCESS_MAILSORTING, ACCESS_MINERAL_STOREROOM)
```
Comment (DaedalusGame) [inline code/game/objects/items/cards_ids.dm]: CL

Comment (Xhuis) [reply code/game/objects/items/cards_ids.dm]: :cl:

Comment (DaedalusGame) [reply code/game/objects/items/cards_ids.dm]: 🅱️

**Comment by davethwave** [comment]
And make the good loot sparser?

**Comment by PsyCommando** [comment]
I'm not sure why you'd nerf sentient minebots that much? Its kind of really boring usually to play as a sentient minebot (pun not intended XD ), so it still should have some incentives.. 

How about you halve the health upgrade on sentient bots if you really must nerf? Or give it better armor instead of health? Or simply not nerf anything directly and make sentient ones run on an internal charge, like cyborgs and mechs?

**Comment by PKPenguin321** [comment]
Because have you ever tried to fight a sentient minebot as a human

**Comment by DaedalusGame** [comment]
just hit it or shoot it with ion shite?
or mix uranium and iron in a beacon?

**Comment by PKPenguin321** [comment]
yes let me just grab the ion rifle and uranium mixing kit out of my bag that every crew member spawns with. How could I forget about those you're totally right great argument

**Comment by DaedalusGame** [comment]
>just hit it

**Comment by PKPenguin321** [comment]
It outright crits you with its melee attack a century before you will come even close to killing it and does enough damage to cause significant slowdown in one or two hits so you can't dodge or outrun it

If you could literally out-damage it by just hitting it do you really think I'd be complaining

**Comment by PsyCommando** [comment]
Ok, so looking at the code, the base minebot does 15 at melee range. (As much as a space carp.) Upgraded it goes up to 22. It has 125 health stock and 170 upgraded.
It also has the following flags : 
`status_flags = CANSTUN|CANKNOCKDOWN|CANPUSH`

As a comparison, a goliath, aka a common lavaland enemy, has 300hp and has 25 as melee attack.. 

With what were you attacking it? Did you wear any sort of protective gear?

**Comment by ChangelingRain** [comment]
It also has the KA, which does 10 at range if you're in any notable atmosphere, and 40 if you're not.
It being stunnable is... technically something that makes it weaker, but also nobody is aware that it's stunnable since it's a simple mob robot and nobody thinks to tase robots.

File: code/modules/mining/minebot.dm [hunk 2]
```diff
@@ -14,84 +15,104 @@
...
+	..()
+	var/t_He = p_they(TRUE)
+	var/t_him = p_them()
+	var/t_s = p_s()
+	if(health < maxHealth)
+		if(health >= maxHealth * 0.5)
+			to_chat(user, "<span class='warning'>[t_He] look[t_s] slightly dented.</span>")
+		else
+			to_chat(user, "<span class='boldwarning'>[t_He] look[t_s] severely dented!</span>")
+	to_chat(user, "<span class='notice'>Using a mining scanner on [t_him] will instruct [t_him] to drop stored ore. <b>[min(0, LAZYLEN(contents) - 1)] Stored Ore</b>\n\
```
Comment (Cyberboss) [inline code/modules/mining/minebot.dm]: max?

**Review by Cyberboss — COMMENTED** [review]
Fine otherwise

**Comment by PsyCommando** [comment]
Yeaah.. But its about the same as a miner with a KA + survival knife. 
Then again, miners can't have 170hp.. I guess an hp tweak might really be a solution in the end..

And yeah its really weird that its stunnable, :/ It doesn't even falls to the ground when stunned with a taser it just stays in place, even if it has the knockdown flag..
It doesn't seem to be vulnerable to what other robots are either.. It might be a bit inconsistent.

**Comment by ChangelingRain** [comment]
Unlike the miner, it doesn't have to switch hands, and cannot be disarmed.

**Pull request merged**